### ISSUE TITLE
Printing a long string causes Flutter to crash

### ISSUE THREAD

**Issue by lukechurch** [flutter/flutter#6758]
## Steps to Reproduce

The below, rather silly, code simulates a real problem that I came across whilst debugging an app.

Run the standard template app with _incrementCounter modified to 

```
void _incrementCounter() {
    setState(() {
      _counter++;
    });

    print ("This is a short string");

    StringBuffer sb = new StringBuffer();
    for (int i = 0; i < 10000; i++) {
      sb.write("This is some text $i");
    }

    print (sb.toString());
  }
```

I.e. Print a short string followed by a very long string.

Run on the iOS simulator. The result is:

```
com.apple.CoreSimulator.SimDevice.B6E113C1-725A-43E9-B4AD-538AD90FC5B2.launchd_sim[12828] (UIKitApplication:com.yourcompany.longString[0x4ee1]: Service exited due to signal: Broken pipe: 13
SpringBoard: Application 'UIKitApplication:com.yourcompany.longString[0x4ee1]' exited abnormally via signal.
assertiond: notify_suspend_pid() failed with error 7
SpringBoard: Reply Error: Connection interrupted
```

## Flutter version details from IntelliJ

<img width="604" alt="screen shot 2016-11-08 at 13 25 36" src="https://cloud.githubusercontent.com/assets/2978182/20100769/e1ebb812-a5b6-11e6-827a-eabb58dc05c7.png">

**Comment by sethladd** [comment]
Curious, does this happen on Android?

**Comment by chinmaygarde** [comment]
I was able to reproduce this crash. It seems there is an issue in when we redirect stdout and stderr output to syslog. Investigating.

**Comment by Hixie** [comment]
As a workaround, consider using `debugPrint` instead of `print`.

**Comment by Hixie** [comment]
https://github.com/flutter/flutter/issues/232 is the bug for moving the debugPrint logic to print itself.

**Comment by Hixie** [comment]
> Curious, does this happen on Android?

Android drops text rather than crashing. It's why we have `debugPrint`, which throttles the text. We should make `print` pluggable so that the machinery that plugs into `debugPrint` instead plugs into `print`.

**Comment by Hixie** [comment]
We also want to make sure we're showing logs from the system, like crash logs and the like.

**Comment by sethladd** [comment]
Customer Lefty hit this.

**Comment by eseidelGoogle** [comment]
I believe a fix was discussed.  @chinmaygarde are you actively working on this?

**Comment by eseidelGoogle** [comment]
@chinmaygarde you mentioned unassigning yourself from this?  Are you still the right owner?

**Comment by chinmaygarde** [comment]
Based on discussions with @cbracken , I have a WIP patch that [attempts to fix this](https://github.com/flutter/engine/pull/3483). However, I am currently in the middle of improving performance in the iOS simulator. Unassigning myself from this ticket for now. If anyone wants to continue working on this before I get back to it, I would be happy to provide help.

**Comment by eseidelGoogle** [comment]
I'm inclined to punt this.  We've seen customers hit this, but I don't think this needs to block M2.

**Comment by cbracken** [comment]
Agreed.

**Comment by cbracken** [comment]
Fixed in flutter/engine#3797. Rolled into the framework in #10859. This will be part of the `0.0.11` alpha release when it's cut.
